processing/src/main/java/io/druid/query/topn/TopNQueryEngine.java

```diff
@@ -97,8 +106,10 @@ public TopNQueryEngine(StupidPool<ByteBuffer> bufferPool)
     final TopNAlgorithmSelector selector = new TopNAlgorithmSelector(cardinality, numBytesPerRecord);
     query.initTopNAlgorithmSelector(selector);
 
-    TopNAlgorithm topNAlgorithm = null;
-    if (selector.isHasDimExtractionFn()) {
+    final TopNAlgorithm topNAlgorithm;
+    if (selector.isHasExtractionFn() && dimension.equals(Column.TIME_COLUMN_NAME)) {
```

Comment:
The `Column.TIME_COLUMN_NAME` is getting proliferated throughout the code quite a bit.  I think we should try to contain it as much as possible.